cmd/setup-gh.go

```diff
@@ -30,6 +30,9 @@ application and service principle, and will configure that application to trust
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
 
+			providers.EnsureAzCli()
```

Comment:
Do we want to consider in the future moving to the azure sdk for go?

Reply:
we have migrated significant chunks of the az interaction to the sdk already. Entra ID app registration management is the limiting factor. The [Graph SDK supports entraid app management](https://learn.microsoft.com/en-us/graph/api/application-post-applications?view=graph-rest-1.0&tabs=http).

onboarding to the graph sdk would require registering as a graph app (most likely first-party). this is likely the best direction forward to consolidate dependencies